p2p/transport/libp2p.go

```diff
@@ -0,0 +1,39 @@
+package transport
```

Comment:
The transport package contained the different "connection types" we have (i.e. empty and TCP).

I'm struggling to understand the code organization, but am thinking that if we have a libP2P transport implementing the transport interface, it should also be a connection type.